frontend/src/bundles/talent-onboarding/store/actions.ts

```diff
@@ -9,8 +9,44 @@ const updateTalentDetails = createAsyncThunk<
     UserDetailsGeneralCustom,
     UserDetailsGeneralCustom,
     AsyncThunkConfig
->(`${sliceName}/update-talent-details`, (registerPayload) => {
-    return registerPayload;
+>(`${sliceName}/update-talent-details`, async (registerPayload, { extra }) => {
+    const { talentOnBoardingApi, fileUploadApi } = extra;
+    const { cv, photo, ...restPayload } = registerPayload;
+    let cvId: string, photoId: string;
+
+    if (cv) {
+        cvId = await fileUploadApi.uploadDocument({ cv });
+        restPayload.cvId = cvId;
+    }
+
+    if (photo) {
+        photoId = await fileUploadApi.uploadImage({ photo });
+        restPayload.photoId = photoId;
+    }
+
+    if ('badges' in registerPayload) {
+        return registerPayload;
+    }
```

Comment:
Why in this case we just return payload ?

Reply:
Hello, this is based off of an earlier version of #344. This has been addressed there in a newer version.